src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -165,20 +161,19 @@ contract GameToken is ERC721BaseToken, WithMinter, IGameToken {
         uint256[] memory values,
         address editor,
         string memory uri,
-        uint64 randomId
+        uint64 subId
     ) external override onlyMinter() notToZero(to) notToThis(to) returns (uint256 id) {
-        uint256 gameId = _mintGame(from, to, randomId, 0);
+        uint256 gameId = _mintGame(from, to, subId, 0);
         uint224 baseId = _extractBaseId(gameId);
 
         if (editor != address(0)) {
             _setGameEditor(gameId, editor, true);
         }
 
         if (assetIds.length != 0) {
-            addAssets(from, gameId, assetIds, values, uri, true);
+            _addAssets(from, baseId, assetIds, values);
         }
-
-        _setTokenURI(gameId, baseId, uri);
+        _metaData[baseId] = uri;
```

Comment:
yes.
Maybe we use the existing GameTokenUpdated event, with `oldId` set to 0 and `newId` with the newly minted Id ?
I was thinking to reuse this event when a game is destroyed as well, similarly setting the `newId` in that case to 0.
Then it will be obvious when parsing events that an `oldId` of 0 = creation and a `newId` of 0 = destruction.